syscall/syscall.c

```diff
@@ -267,23 +274,57 @@ OE_DEFINE_SYSCALL2(SYS_flock)
     return oe_flock(fd, operation);
 }
 
-OE_DEFINE_SYSCALL2(SYS_fstat)
+OE_WEAK OE_DEFINE_SYSCALL2(SYS_fstat)
 {
     oe_errno = 0;
     const int fd = (int)arg1;
     struct oe_stat_t* const buf = (struct oe_stat_t*)arg2;
     return oe_fstat(fd, buf);
 }
 
-OE_DEFINE_SYSCALL1(SYS_fsync)
+OE_WEAK OE_DEFINE_SYSCALL4(SYS_fstatat)
+{
+    oe_errno = 0;
+    long ret = -1;
+    int dirfd = (int)arg1;
+    const char* pathname = (const char*)arg2;
+    struct oe_stat_t* buf = (struct oe_stat_t*)arg3;
```

Comment:
```suggestion
    struct oe_stat_t* buffer = (struct oe_stat_t*)arg3;
```
Although some name closer to "stat" would be more clear